src/source.ts

```diff
@@ -127,82 +128,98 @@ export class Source {
   }
 
   /**
-   * Create or update a security identity. See [Adding a Single Security Identity](https://docs.coveo.com/en/167) and [Security Identity Models](https://docs.coveo.com/en/139).
-   * @param securityProviderId
-   * @param securityIdentity
-   * @returns
+   * @deprecated use `identity.createSecurityIdentity`
+   *
+   * See {@link Source.identity}
    */
   public createSecurityIdentity(
     securityProviderId: string,
     securityIdentity: SecurityIdentityModel
   ) {
-    return this.platformClient.pushApi.createOrUpdateSecurityIdentity(
+    console.log('This method has been deprecated');
+    console.log(
+      'Use `source.identity.createOrUpdateSecurityIdentity()` instead'
+    );
```

Comment:
I would not put any logs personally. If someone use it 'in prod' it can create noise.
If we had a 'logger level' configuration, sure maybe we could have, but presently, nah.

Reply:
I am in favour of that noise to incite users to use the appropriate method.